### PR TITLE
refactor(get-starknet): add get-starknet error handling

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
This PR is adding a error handling on get-starknet

Base on PR #374 , as the Snap is expecting to throw a error that contains the get-starknet error code

therefore on get-starknet side, we should have a method to extract the error and format into a new error with defined message, which will be `createStarkError`

A usage example can be found in RPC `switch-network`

### CODE REVIEW

File: packages/get-starknet/src/utils/object.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Verify if the given property is exist in the object or not.
```
Comment (khanti42) [inline packages/get-starknet/src/utils/object.ts:2]:
```suggestion
 * Verify if the given property exists in the object or not.
```

File: packages/get-starknet/src/utils/object.test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    },
+  };
+
+  it.each(['p1.p2.p3'])('returns true if the property "%s" exist in the object', (property: string) => {
+    const result = isset(obj, property);
+
+    expect(result).toBe(true);
+  });
+
+  it('returns false if the property is not exist in the object', () => {
```
Comment (khanti42) [inline packages/get-starknet/src/utils/object.test.ts:19]:
```suggestion
  it('returns false if the property does not exist in the object', () => {
```

File: packages/get-starknet/src/utils/object.test.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  const obj = {
+    p1: {
+      p2a: 1,
+      p2: {
+        p3: 4,
+      },
+    },
+  };
+
+  it.each(['p1.p2.p3'])('returns true if the property "%s" exist in the object', (property: string) => {
```
Comment (khanti42) [inline packages/get-starknet/src/utils/object.test.ts:13]:
```suggestion
  it.each(['p1.p2.p3'])('returns true if the property "%s" exists in the object', (property: string) => {
```

File: packages/get-starknet/src/utils/error.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  readonly code: number;
+
+  constructor(message: string, errorCode: number) {
+    super(message);
+    this.code = errorCode;
+  }
+}
+
+/**
+ * Create Custom Error base on the given error code to map with the Wallet API error.
```
Comment (khanti42) [inline packages/get-starknet/src/utils/error.ts:35]:
```suggestion
 * Create Custom Error based on the given error code to map with the Wallet API error.
```

File: packages/get-starknet/src/utils/error.test.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  });
+
+  it('returns default error code and message if the error code is undefined', () => {
+    const error = createStarkError(undefined);
+    expect(error).toBeInstanceOf(CustomError);
+    expect(error.message).toStrictEqual(defaultErrorMessage);
+    expect(error.code).toStrictEqual(defaultErrorCode);
+  });
+
+  it('returns default error code and message if the error code is not exist in the mapping', () => {
```
Comment (khanti42) [inline packages/get-starknet/src/utils/error.test.ts:23]:
```suggestion
  it('returns default error code and message if the error code does not exist in the mapping', () => {
```

**Review by khanti42 — CHANGES_REQUESTED** [review]

**Review by khanti42 — CHANGES_REQUESTED** [review]
As per discussion elsewhere, let's remove object.test.ts and object.ts as we don't use them

File: packages/get-starknet/src/utils/object.test.ts (hunk 3 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/src/utils/object.test.ts:13]: the file removed

File: packages/get-starknet/src/utils/object.test.ts (hunk 2 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/src/utils/object.test.ts:19]: the file removed

File: packages/get-starknet/src/utils/object.ts (hunk 1 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/src/utils/object.ts:2]: the file removed

**Review by khanti42 — APPROVED** [review]
LGTM

**Pull request merged**